content/scripts/build.js

```diff
@@ -1610,6 +1612,69 @@ class Builder {
           }
         });
 
+        if (this.options.fixFlaws) {
+          // For flaws that can be changed, change the source itself.
+
+          // Copy of the original raw HTML that we're going to (potentially)
+          // repeatedly do string replaces on.
+          let newRawHtml = rawHtml;
+          flaws
+            .filter((flaw) => {
+              // Only interested in 'MacroRedirectedLinkError' flaws because
+              // them we can do something about.
+              // The flaw might be from a prerequisite, if so, leave it alone
+              // See https://github.com/mdn/yari/issues/772
+              return (
+                flaw.name === "MacroRedirectedLinkError" &&
+                (!flaw.filepath || path.dirname(flaw.filepath) === folder)
+              );
+            })
+            .forEach((flaw, i) => {
+              if (!rawHtml.includes(flaw.macroSource)) {
+                throw new Error(
+                  `rawHtml doesn't contain macrosSource (${flaw.macroSource})`
+                );
+              }
+              const newMacroSource = flaw.macroSource.replace(
+                flaw.redirectInfo.current,
+                flaw.redirectInfo.suggested
+              );
+              newRawHtml = newRawHtml.replace(flaw.macroSource, newMacroSource);
+
+              // If the flaw could be fixed, it's no longer a flaw. So remove
+              // it from the array.
+              // This means you could run the CLI like this:
+              // `build ... --fix-flaws --flaw-levels="macros:error"`
+              // it would only complain about this that couldn't be fixed.
+              if (newRawHtml !== rawHtml) {
+                flaws.splice(i, 1);
+              }
+            });
+
+          if (newRawHtml !== rawHtml) {
+            // It was improved!!
+            if (this.options.fixFlawsVerbose) {
+              console.log(`\nIn ${folder}...`);
+              printBasicDiff(rawHtml, newRawHtml);
+            }
+            if (this.options.fixFlawsDryRun) {
+              console.log(chalk.yellow("Not changing any files in dry-run!"));
```

Comment:
Does it make any sense to use `--fix-flaws-dry-run` without also using `fix-flaws-verbose`? Perhaps `--fix-flaws-dry-run` should always imply `fix-flaws-verbose`?

In any case, what do you think about adding a bit more detail in the message, something like: 
```suggestion
              console.log(chalk.yellow(`Would have modified "${rawHtmlFilepath}", if this was not a dry run.`));
```
since it makes `--fix-flaws-dry-run` more informative when `fix-flaws-verbose` is not used as well.